scripts/composer/Setup.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+namespace DrupalProject\composer;
+
+use Composer\Json\JsonFile;
+use Composer\Script\Event;
+use Symfony\Component\Filesystem\Filesystem;
+
+class Setup {
+
+  public static function setup(Event $event) {
+    $jsonFile = new JsonFile($event->getComposer()
+      ->getConfig()
+      ->getConfigSource()
+      ->getName());
+    $config = $jsonFile->read();
+    $fs = new Filesystem();
+    $drupalRoot = $event->getIO()->ask('<info>Customize Drupal root path?</info> [<comment>web</comment>]? ', 'web');
+    if ($drupalRoot !== 'web') {
+      $drupalRoot = rtrim($drupalRoot, '/');
+      $gitIgnore = file_get_contents('.gitignore');
+      $gitIgnore = preg_replace('/' . preg_quote('web/', '/') . '/', $drupalRoot . '/', $gitIgnore);
+      if (isset($config['extra']['installer-paths'])) {
+        $installer_paths = [];
+        foreach ($config['extra']['installer-paths'] as $path => $spec) {
+          $newPath = preg_replace('/' . preg_quote('web/', '/') . '/', $drupalRoot . '/', $path);
+          $installer_paths[$newPath] = $spec;
+        }
+        $config['extra']['installer-paths'] = $installer_paths;
+      }
+      $fs->dumpFile('.gitignore', $gitIgnore);
```

Comment:
Move this before line 23. The logical flow would be to save the new gitignore just after the preg_replace on the gitIgnore lines, and not after the installer_paths block.